csharp/src/Ice/IncomingRequestFrame.cs

```diff
@@ -54,17 +54,15 @@ public IncomingRequestFrame(Protocol protocol, ArraySegment<byte> data)
             Context = istr.ReadContext();
             Payload = Data.Slice(istr.Pos);
             (int size, Encoding encoding) = istr.ReadEncapsulationHeader();
-            if (protocol == Protocol.Ice1 && size + 4 != Payload.Count)
-            {
-                // The payload holds an encapsulation and the encapsulation must use up the full buffer with ice1.
-                // "4" corresponds to fixed-length size with the 1.1 encoding.
-                throw new InvalidDataException($"invalid request encapsulation size: {size}");
-            }
-            else
+
+            if (protocol == Protocol.Ice1 && (4 + size != Payload.Count))
```

Comment:
Should we return the sizeLenght from `ReadEncapsulationHeader` and do this checks for both protocols using sizeLength istead of 4

Reply:
I think we should usr istr.Pos in this case: `istr.Pos - 2 + size == Payload.Count`. For another PR!